tests/unit/unit-walker.cpp

```diff
@@ -115,6 +161,730 @@ TEST_P(UnitWalkerTests, FunSubstitution)
   EXPECT_EQ(fy, iw.visit(fx));
 }
 
+TEST_P(UnitWalkerTests, SimpleTree)
+{
+  // traversal of simple formula using TreeWalker, test that cache gets built up as expected
+
+  SolverConfiguration sc = GetParam();
+  if (sc.is_logging_solver) {
+    // test only valid for logging solver
+
+    Term x = s->make_symbol("x", bvsort);
+    Term xp1 = s->make_term(BVAdd, x, s->make_term(1, bvsort));
+
+    UnorderedTermPairMap UndMap;
+    TreeWalker iw(s, false, &UndMap);
+
+    pair <Term, vector<int>> p1;
+    p1 = iw.visit(xp1);
+    Term xp11 = p1.first;
+    EXPECT_EQ(xp1, xp11);
+
+    // expected_map:
+    // x+1: <x+1, []>
+    // x: <x+1, [0]>
+    // 1: <x+1, [1]>
+    map<Term, pair<Term, vector<int>>> expected_map;
+    pair <Term, vector<int>> n0;
+    n0.first = xp1;
+    vector<int> n0path;
+    n0.second = n0path;
+    expected_map[xp1] = n0;
+    pair <Term, vector<int>> n1;
+    n1.first = xp1;
+    vector<int> n1path = {0};
+    n1.second = n1path;
+    expected_map[x] = n1;
+    pair <Term, vector<int>> n2;
+    n2.first = xp1;
+    vector<int> n2path = {1};
+    n2.second = n2path;
+    expected_map[s->make_term(1, bvsort)] = n2;
+
+    // test equivalence of expected_map and cache populated during the traversal
+    for (auto const& x : UndMap){
+      // testing equivalence of topmost node (first element in pair)
+      EXPECT_EQ(x.second.first, expected_map[x.first].first);
+      ASSERT_EQ(x.second.second.size(), expected_map[x.first].second.size());
+      // testing path equivalence
+      for (int i = 0; i < x.second.second.size(); i++){
+        EXPECT_EQ(x.second.second[i], expected_map[x.first].second[i]);
+      }
+    }
```

Comment:
plz make a helper function for this code, and call it in every test, instead of inlining it.

Reply:
double check on placement w/in file and my instantiation